rust/lance-index/src/scalar/inverted.rs

```diff
@@ -391,50 +471,92 @@ impl InvertedList {
             let token_col = batch[TOKEN_ID_COL].as_primitive::<datatypes::UInt32Type>();
             let row_ids_col = batch[ROW_ID].as_list::<i32>();
             let frequencies_col = batch[FREQUENCY_COL].as_list::<i32>();
+            let scores_col = batch[SCORE_COL].as_list::<i32>();
 
-            for ((&token_id, row_ids), frequencies) in token_col
+            for (((&token_id, row_ids), frequencies), scores) in token_col
                 .values()
                 .iter()
                 .zip(row_ids_col.iter())
                 .zip(frequencies_col.iter())
+                .zip(scores_col.iter())
             {
                 let row_ids = row_ids.unwrap();
                 let frequencies = frequencies.unwrap();
-                let row_ids = row_ids.as_primitive::<UInt64Type>().values();
-                let frequencies = frequencies.as_primitive::<UInt64Type>().values();
-                let list = row_ids
-                    .iter()
-                    .cloned()
-                    .zip(frequencies.iter().cloned())
-                    .collect_vec();
-                inverted_list.insert(token_id, list);
+                let scores = scores.unwrap();
+                let row_ids = row_ids.as_primitive::<UInt64Type>().values().to_vec();
+                let frequencies = frequencies.as_primitive::<Float32Type>().values().to_vec();
+                let scores = scores.as_primitive::<Float32Type>().values().to_vec();
+                inverted_list.insert(token_id, PostingList::new(row_ids, frequencies, scores));
             }
         }
 
         Ok(Self { inverted_list })
     }
 
+    fn remap(&mut self, mapping: &HashMap<u64, Option<u64>>) {
+        for list in self.inverted_list.values_mut() {
+            let mut new_row_ids = Vec::new();
+            let mut new_freqs = Vec::new();
+            let mut new_scores = Vec::new();
+
+            for i in 0..list.len() {
+                let row_id = list.row_ids[i];
+                let freq = list.frequencies[i];
+                let score = list.scores[i];
+
+                match mapping.get(&row_id) {
+                    Some(Some(new_row_id)) => {
+                        new_row_ids.push(*new_row_id);
+                        new_freqs.push(freq);
+                        new_scores.push(score);
+                    }
+                    Some(None) => continue,
+                    None => {
+                        new_row_ids.push(row_id);
+                        new_freqs.push(freq);
+                        new_scores.push(score);
+                    }
```

Comment:
```suggestion
                    _ => continue,
```

The `None` case shouldn't ever happen (the `mapping` should contain an entry for every old row id) but if it did happen I think a safer error behavior would be to treat it the same as `Some(None)`.